tests/integration_tests/conftest.py

```diff
@@ -459,8 +468,27 @@ def _generate_profile_report() -> None:
     log.info(command, "final.stats")
 
 
+# https://docs.pytest.org/en/stable/reference/reference.html#pytest.hookspec.pytest_sessionstart
+def pytest_sessionstart(session) -> None:
+    """do session setup"""
+    global _SESSION_CLOUD
+    try:
+        _SESSION_CLOUD = get_session_cloud()
+        setup_image(_SESSION_CLOUD)
+    except Exception as e:
+        pytest.exit(
```

Comment:
My only concern left is cleanup. Will `pytest_sessionfinish()` run if this line gets hit? If not, we should have cleanup here too.

Reply:
> Will pytest_sessionfinish() run if this line gets hit? If not, we should have cleanup here too.

No, added.